File: app/src/main/java/ru/nikshlykov/englishwordsapp/SubgroupDao.java
package ru.nikshlykov.englishwordsapp;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface SubgroupDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    long insert(Subgroup subgroup);

    @Update
    int update(Subgroup subgroup);

    @Delete
    int delete(Subgroup subgroup);

    @Query("SELECT * FROM Subgroups WHERE _id = :id")
    Subgroup getSubgroupById(long id);

    @Query("SELECT * FROM Subgroups WHERE groupId = :groupId")
    Subgroup[] getSubgroupsFromGroup(long groupId);

    @Query("SELECT * FROM Subgroups WHERE IsStudied = 1")
    Subgroup[] getStudiedSubgroups();
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/daos/SubgroupDao.kt
package ru.nikshlykov.englishwordsapp.db.daos

import androidx.room.*
import ru.nikshlykov.englishwordsapp.db.models.Group
import ru.nikshlykov.englishwordsapp.db.models.Subgroup

@Dao
interface SubgroupDao {

  @Insert(onConflict = OnConflictStrategy.IGNORE)
  suspend fun insert(subgroup: Subgroup): Long

  @Update
  suspend fun update(subgroup: Subgroup): Int

  @Delete
  suspend fun delete(subgroup: Subgroup): Int

  @Query("SELECT * FROM Subgroups WHERE _id = :id")
  suspend fun getSubgroupById(id: Long): Subgroup

  @Query("SELECT * FROM Subgroups WHERE groupId = :groupId")
  suspend fun getSubgroupsFromGroup(groupId: Long): List<Subgroup>

  @Query("SELECT * FROM Subgroups ORDER BY _id LIMIT 1")
  suspend fun subgroupWithMinId(): Subgroup

  @Query("SELECT * FROM Subgroups WHERE groupId = ${Group.GROUP_FOR_NEW_SUBGROUPS_ID}")
  suspend fun getCreatedByUserSubgroups(): List<Subgroup>
}